### ISSUE TITLE
vi mode prints errors in Emacs when TERM=dumb

### ISSUE THREAD

**Issue by meatcar** [fish-shell/fish-shell#5806]
Running any command in Emacs when vi mode is enabled prints a lot of errors to stdout.

```
$ fish --version
fish, version 3.0.2
$ uname -a
Linux hostname 5.0.7-arch1-1 #4 SMP PREEMPT Wed Apr 10 21:36:08 +08 2019 x86_64 GNU/Linux
$ echo $TERM
dumb
$ cat ~/.config/fish/config.fish
set fish_greeting
fish_vi_key_bindings
```

Running <kbd>M-x</kbd> `shell`:
```
bind: Key with name “right” does not have any mapping
bind: Key with name “left” does not have any mapping
bind: Key with name “ppage” does not have any mapping
bind: Key with name “npage” does not have any mapping
bind: Key with name “btab” does not have any mapping
bind: Key with name “down” does not have any mapping
bind: Key with name “up” does not have any mapping
bind: Key with name “f1” does not have any mapping
bind: Key with name “right” does not have any mapping
bind: Key with name “left” does not have any mapping
bind: Key with name “ppage” does not have any mapping
bind: Key with name “npage” does not have any mapping
bind: Key with name “btab” does not have any mapping
bind: Key with name “down” does not have any mapping
bind: Key with name “up” does not have any mapping
bind: Key with name “f1” does not have any mapping
bind: Key with name “right” does not have any mapping
bind: Key with name “left” does not have any mapping
bind: Key with name “ppage” does not have any mapping
bind: Key with name “npage” does not have any mapping
bind: Key with name “btab” does not have any mapping
bind: Key with name “down” does not have any mapping
bind: Key with name “up” does not have any mapping
bind: Key with name “f1” does not have any mapping
$ 
```

`fish_vi_key_bindings 2>/dev/null` fixes the issue.

**Comment by faho** [comment]
Yeah, you have an old version of the `fish_vi_key_bindings` function. This was explicitly fixed in #5028.

Also, you *really* don't want TERM=dumb. Because those errors actually mean that those keys won't work. In addition, TERM=dumb does basically _nothing_ - no color, no cursor movement, no nothing.

**Comment by meatcar** [comment]
Issue is confirmed fixed in master, my bad. Thanks!

Sadly getting emacs to not set TERM=dumb for every command it runs is out of my control.
